src/frontend/hooks/server/projects.ts

```diff
@@ -159,3 +160,63 @@ export function useGetOwnRole() {
     },
   });
 }
+
+export function useGetRemoteArchives() {
+  const {projectId, projectApi} = useActiveProject();
+
+  return useQuery({
+    queryKey: [REMOTE_ARCHIVE, projectId, PROJECT_MEMBERS_KEY],
+    queryFn: async () => {
+      const members = await projectApi.$member.getMany();
+      const filteredMembers = members.filter(
+        member => member.deviceType === 'selfHostedServer',
+      );
+      return filteredMembers;
+    },
+  });
+}
+
+export function useAddRemoteArchive() {
+  const {projectApi} = useActiveProject();
+  const queryClient = useQueryClient();
+  return useMutation({
+    mutationFn: async (normalizedUrl: string) => {
+      return projectApi.$member.addServerPeer(normalizedUrl);
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({queryKey: [REMOTE_ARCHIVE]});
+    },
+  });
+}
+
+export function useFindRemoteArchive({url}: {url?: string}) {
+  return useQuery({
+    queryFn: async () => {
+      if (!url) throw new Error('no url');
+      const response = await fetch(url);
+
+      if (response.status !== 200) {
+        throw new Error('Server should return a 200');
+      }
+
+      const responseJson = await response.json();
+
+      if (
+        responseJson &&
+        typeof responseJson === 'object' &&
+        'data' in responseJson &&
+        responseJson.data &&
+        typeof responseJson.data === 'object' &&
+        'name' in responseJson.data &&
+        responseJson.data.name &&
+        typeof responseJson.data.name === 'string'
+      ) {
+        return responseJson.data.name;
+      } else {
+        throw new Error('Server responded with unexpected data');
```

Comment:
We could have just thrown a generic error with no message, but this is more useful for debugging purposes.

So no, we are not using it, but I thought it would be more helpful for debugging puposes to create this message.